### DISCUSSION TITLE
ReactPlay template update

### DISCUSSION THREAD

**Discussion by koustov** [reactplay/react-play#201] [Ideas]
Hello All,
I am looking for suggestions for improving plop template for a better user experience. I have a few things in mind
- Style template: the command line should allow the user to select from CSS/SCSS/None template and deploy accordingly
- Readme: Play read me should automatically be updated with boilerplate information. Something like name, description, language used, blog info, etc.
- Tag Suggestion: Allow user to select from existing tag instead of creating a new one.

Whats in your mind?

**Could be a long term goal: Leverage UI for add/edit a play**

**Comment by atapas** [comment]
These are great suggestions. Let's create issues for them and tackle them one by one.

> Style template: the command line should allow the user to select from CSS/SCSS/None template and deploy accordingly

Create an issue. It should create the CSS/SCSS file based on input. We should use camelCase on the play name to decide the file name. Also, we should change the component template to import the CSS/SCSS file if they exist.

> Readme: Play read me should automatically be updated with boilerplate information. Something like name, description, language used, blog info, etc.

It is good to have issue. The primary purpose of the Readme is to provide the technical details when the creator doesn't provide a blog link(https://github.com/atapas/react-play/issues/200). I would suggest creating an issue explaining what we need to add as a part template and what fashion(table?).

> Tag Suggestion: Allow user to select from existing tag instead of creating a new one.

Agree. It will stop the tag flooding. Create issue.

I have one more to add: In the component template, let's add a unique class name and the existing class. The unique class name can be of the kebab case based on the play name. Right here(an example)

https://github.com/atapas/react-play/blob/734065b318b74e04c3b03b38ef3f41d1a6003bbd/src/plays/states/States.jsx#L42

Play creator can use this unique class name to achieve CSS Specificity in the CSS/SCSS file. What do you think?


So we are talking about 4 new issues and the need to create them.

**Reply by atapas** [reply]
@koustov Do we have all the issues created? If you could provide the links here, will lock this discussion and mark as closed.

**Reply by MOHAMED-EHAB-DEV** [reply]
Hi Tapas

I am happy to work with you and @koustov on it
